### PR TITLE
Production API app.

### PR DESCRIPTION
**Pull request by ride90**
SDBELGA-169
SDBELGA-168

- [x] prod api app structure
- [x] versioning
- [x] serve assets
- [x] items endpoint
- [x] desks endpoint
- [x] planning related endpoints
    * [x] events endpoint
    * [x] planning endpoint
    * [x] assignments endpoint
- [x] contacts endpoint
- [x] users endpoint

@petrjasek @jerome-poisson for now prod api app does not have any auth, this is something we can implement later. I did not cover all this read only endpoints with tests, will do it after we integrate auth, now there is not much to test :) The same with a documentation, I'll add docs on later stages.

P.S. Regarding versioning, for now it's just a `v1` in the url, I think it's fine for now. IMO there is no sense to care now about a flexible structure inside prod_api app to cover possible future api versions (v1, v2, v3 subfolders etc). V2 might be a graphql or something else in a different app, so I think no sense to deal with it now.

### CODE REVIEW

File: prod_api/app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    app.sentry = SuperdeskSentry(app)
+
+    return app
+
+
+if __name__ == '__main__':
+    host = '0.0.0.0'
+    port = int(os.environ.get('PORT', '5500'))
+    app = get_app()
+    app.run(host=host, port=port, debug=True, use_reloader=True)
```
Comment (jerome-poisson) [inline prod_api/app/__init__.py]: `debug` should not be `True` by default, we should use a environment variable for developers for that. Same with reloader.

File: prod_api/desks/resource.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class DesksResource(Resource):
+    url = 'desks'
+    item_url = item_url
+    item_methods = ['GET']
+    resource_methods = ['GET']
+    datasource = {
+        'source': 'desks',
+        'default_sort': [('name', 1)],
+        'projection': {
+            # NOTE: since schema is not definted here, setting up a projection explicitly is required,
```
Comment (jerome-poisson) [inline prod_api/desks/resource.py]: type: `definted`

File: prod_api/items/resource.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    url = 'items'
+    item_url = item_url
+    item_methods = ['GET']
+    resource_methods = ['GET']
+    datasource = {
+        'source': 'archive',
+        'search_backend': 'elastic',
+        'default_sort': [('_updated', -1)],
+        'projection': {
+            # NOTE: since schema is not definted here, setting up a projection explicitly is required,
```
Comment (jerome-poisson) [inline prod_api/items/resource.py]: same type: `definted`

File: prod_api/__init__.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014, 2015 Sourcefabric z.u. and contributors.
```
[the remark this answers is not in the document]
Comment (petrjasek) [reply prod_api/__init__.py]: afaik it should be enough to say when the copyright started, so when file was created, and it should last afterwards. so new files should have copyright 2019

File: prod_api/app/__init__.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            pass
+
+    app.sentry = SuperdeskSentry(app)
+
+    return app
+
+
+if __name__ == '__main__':
+    host = '0.0.0.0'
+    port = int(os.environ.get('PORT', '5500'))
```
[the remark this answers is not in the document]
Comment (petrjasek) [reply prod_api/app/__init__.py]: I think `PORT` is what you get from honcho

[the remark this answers is not in the document]
Comment (ride90) [reply prod_api/app/__init__.py]: I'll change it to `SD_PRODAPI_PORT`. I think `PORT` is not from honcho or at least I can run prod api without honcho (as I do all the time for development).

[the remark this answers is not in the document]
Comment (petrjasek) [reply prod_api/app/__init__.py]: yep but it you run it with honcho with other apps like superdesk and content api you will get different `PORT` for each

[the remark this answers is not in the document]
Comment (ride90) [reply prod_api/app/__init__.py]: With honcho we'll run a production api with gunicorn and each process will get their own `PORT`, so it's fine.
For development I think having `PORT` is also fine. For me it's not an issue, I run Content API, Production API and Superdesk manually and can overwrite `PORT` for every process if It's needed. 
In general If we want to get rid of generic name like `PORT`, we should do it also for content api, superdesk app and production api. Should I create a ticket for it? @jerome-poisson @petrjasek

File: prod_api/app/__init__.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        except AttributeError:
+            pass
+
+    app.sentry = SuperdeskSentry(app)
+
+    return app
+
+
+if __name__ == '__main__':
+    host = '0.0.0.0'
```
[the remark this answers is not in the document]
Comment (ride90) [reply prod_api/app/__init__.py]: @jerome-poisson I think this part is only for development usage, I'm sure that we'll run prod api with `honcho -> gunicorn`, so we can specify it in `gunicorn_config.py` for prod api if we need it.

File: prod_api/app/__init__.py (hunk 1 above)
Comment (ride90) [reply prod_api/app/__init__.py]: This is for development, so maybe having debug and auto reload is ok in this case.

File: prod_api/app/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jerome-poisson) [reply prod_api/app/__init__.py]: well it's not that important, it just may cause trouble if an other software is using the same environment variable. Maybe a low priority ticket could be created indeed.

File: prod_api/app/__init__.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jerome-poisson) [reply prod_api/app/__init__.py]: if we use it in dev only, would not `127.0.0.1` be more appropriate? I don't see any reason to open it to the world.

File: prod_api/app/__init__.py (hunk 1 above)
Comment (jerome-poisson) [reply prod_api/app/__init__.py]: :+1:

**Pull request merged**